### PR TITLE
[fix](stmt):fix CreateTableStmt toSql placed comment in wrong place

### LINKED ISSUE #27474: [Bug] CreateTableStmt‘s  toSql put comment in the wrong place
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

- 1.2.7.1
- 2.0.x

### What's Wrong?

- CreateTableStmt parsedSql generated by function toSql() put comment in wrong place
image

- CreateTableStmt engineName will be upperCase，when user does't have Create_priv and engine in CreateTable DDL String is upperCase 
image

### What You Expected?

- CreateTableStmt toSql get correct dll
- engineName will ignore case

### How to Reproduce?

1. In AuditPlugin, if user has no create_priv, the AuditEvent.stmt will get wrong parsed sql becauseof comment in wrong place;
2. In AuditPlugin, if user has no create_priv and user's create table ddl's ENGINE is OLAP, engineName in the instance of CreateTableStmt will be upperCase;

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by htyoung**
## Proposed changes

Issue Number: close #27474 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java [hunk 1]
```diff
@@ -715,12 +714,16 @@ public String toSql() {
             }
         }
         sb.append("\n)");
-        sb.append(" ENGINE = ").append(engineName);
+        sb.append(" ENGINE = ").append(engineName.toUpperCase());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:717]: use toLowerCase

Comment (htyoung) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java]: It has been modified to toLowerCase, and the code format has been adjusted using the project's IntelliJ-code-format.

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by hf200012 — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by hf200012 — APPROVED** [review]

**Pull request merged**